Tasks/ArchiveFiles/archivefiles.ts

```diff
@@ -0,0 +1,273 @@
+    }
+    return handleExecResult(tar.execSync(getOptions()), archive);
+}
+
+function handleExecResult(execResult, archive) {
+    if (execResult.code != tl.TaskResult.Succeeded) {
+        var message = 'Archive creation failed for archive file: ' + archive + ' See log for details.';
+        failTask(message);
+    }
+}
+
+function failTask(message: string) {
+    tl.setResult(tl.TaskResult.Failed, message);
+}
+
+/**
+ * Windows only
+ * standard gnu-tar extension formats with recognized auto compression formats
+ * https://www.gnu.org/software/tar/manual/html_section/tar_69.html
+ *   
+ * Computes the name of the tar to use inside a compressed tar.
+ * E.g. foo.tar.gz is expected to have foo.tar inside
+ */
+function computeTarName(archiveName : string) : string {
+    
+    var lowerArchiveName = archiveName.toLowerCase();
+
+    //standard full extensions
+    //                     gzip        xz        bzip2
+    var fullExtensions = ['.tar.gz', '.tar.xz', '.tar.bz2'];
+    for(var i=0; i<fullExtensions.length; i++){
+        if(lowerArchiveName.endsWith(fullExtensions[i])){
+            return archiveName.substring(0, archiveName.lastIndexOf('.'));
+        }
+    }
+
+    //standard abbreviated extensions
+    //                 gzip    gzip    bzip2   bzip2   bzip2   xz
+    var extensions = ['.tgz', '.taz', '.tz2', 'tbz2', '.tbz', '.txz'];
+    for(var i=0; i<extensions.length; i++){
+        if(lowerArchiveName.endsWith(extensions[i])){
+            return archiveName.substring(0, archiveName.lastIndexOf('.')) + '.tar';
+        }
+    }
+
+    // if 7zip falls down here, or if a .tz2 file is encountered, there is occasionally strange
+    // behavior resulting in archives that are correct but are renamed
+    // e.g. test.tz2 has test inside of it instead of test.tar (7z decides to rename it)
+    // e.g. test_tgz has an outer name of test_tgz.gz (7z decides to add the .gz to it).
+
+    //non standard name
+
+    //foo.tar.anything --> foo.tar
+    if (lowerArchiveName.lastIndexOf('.tar.') > 0){
+        return archiveName.substring(0, lowerArchiveName.lastIndexOf('.tar.') + 4);
+    }
+    // foo.anything --> foo.tar
+    else if (lowerArchiveName.lastIndexOf('.') > 0){
+        return archiveName.substring(0, lowerArchiveName.lastIndexOf('.'))+'.tar';
+    }
+    // foo --> foo.tar
+    return lowerArchiveName + '.tar';
+}
+
+//ensure output folder exists
+var destinationFolder = path.dirname(archiveFile);
+tl.debug("Checking for archive destination folder:"+destinationFolder);
+if(!tl.exist(destinationFolder)){
+    tl.debug("Destination folder does not exist, creating:"+destinationFolder);
+    tl.mkdirP(destinationFolder);
+}
+
+if(win) { // windows only
+    if(archiveType == "default"){ //default is zip format
+        sevenZipArchive(archiveFile, "zip", files);
+    } else if(archiveType == "tar"){
+        var tarCompression: string = tl.getInput('tarCompression', true);
+        if(tarCompression == "none"){
```

Comment:
Actually, don't need to change this since I only change the translated string to 'None'